### PR TITLE
Add idp group resources to terraform provider

### PR DESCRIPTION
**Pull request by ryan-riddle-opal**
## Description of the change
Adding idp group mapping to terraform. Tested locally to confirm read, create and update are working.

> Write a brief description of your change here. 

## Checklist

- [x] I performed a self-review of my code
- [x] I manually tested my code change (please list details in description)
- [ ] I added unit tests 
- [ ] I updated the changelog
- [x] I updated the public facing docs

### CODE REVIEW

File: terraform_overlay.yaml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 overlay: 1.0.0
 info:
-  title: Overlay ./openapi_original.yaml => ./openapi.yaml
+  title: Overlay .speakeasy/temp/remote_openapi.yaml => ./openapi.yaml
```
Comment (ryan-riddle-opal) [inline terraform_overlay.yaml:3]: this seems wrong

File: .envrc.local.swp
Comment (ryan-riddle-opal) [inline .envrc.local.swp:1]: will remove

**Review by giulio-opal — COMMENTED** [review]
@amruth-opal @andrewsy-opal would you be able to review and unblock Ryan? Faire has been asking for this feature :)

**Comment by ryan-riddle-opal** [comment]
Looks like I need rebase and test again. I'll do that today and ping when it is ready for review

**Comment by ryan-riddle-opal** [comment]
was easier to just start another branch instead of dealing with all the conflicts
new branch: https://github.com/opalsecurity/terraform-provider-opal/pull/110

**Pull request closed without merging**